crates/primitives/src/transaction/signature.rs

```diff
@@ -222,22 +220,91 @@ pub const fn extract_chain_id(v: u64) -> alloy_rlp::Result<(bool, Option<u64>)>
     }
 }
 
+impl EncodableSignature for Signature {
+    fn from_rs_and_parity<
+        P: TryInto<alloy_primitives::Parity, Error = E>,
+        E: Into<alloy_primitives::SignatureError>,
+    >(
+        r: U256,
+        s: U256,
+        parity: P,
+    ) -> Result<Self, alloy_primitives::SignatureError> {
+        let parity: alloy_primitives::Parity = parity.try_into().map_err(Into::into)?;
+        let signature = Self { r, s, odd_y_parity: parity.y_parity() };
+
+        Ok(signature)
+    }
+
+    fn r(&self) -> U256 {
+        self.r
+    }
+
+    fn s(&self) -> U256 {
+        self.s
+    }
+
+    fn v(&self) -> alloy_primitives::Parity {
+        alloy_primitives::Parity::NonEip155(self.odd_y_parity)
+    }
+
+    fn with_parity<T: Into<alloy_primitives::Parity>>(self, parity: T) -> Self {
+        Self { r: self.r, s: self.s, odd_y_parity: parity.into().y_parity() }
+    }
+}
+
+/// A signature with full parity included.
+// TODO: replace by alloy Signature when there will be an easy way to instantiate them.
+pub(crate) struct SignatureWithParity {
+    /// The R field of the signature; the point on the curve.
+    r: U256,
+    /// The S field of the signature; the point on the curve.
+    s: U256,
+    /// Signature parity
+    parity: Parity,
+}
+
+impl SignatureWithParity {
+    /// Creates a new [`SignatureWithParity`].
+    pub(crate) const fn new(r: U256, s: U256, parity: Parity) -> Self {
+        Self { r, s, parity }
+    }
+}
+
+impl EncodableSignature for SignatureWithParity {
+    fn from_rs_and_parity<
+        P: TryInto<Parity, Error = E>,
+        E: Into<alloy_primitives::SignatureError>,
+    >(
+        r: U256,
+        s: U256,
+        parity: P,
+    ) -> Result<Self, alloy_primitives::SignatureError> {
+        Ok(Self { r, s, parity: parity.try_into().map_err(Into::into)? })
+    }
+
+    fn r(&self) -> U256 {
+        self.r
+    }
+
+    fn s(&self) -> U256 {
+        self.s
+    }
+
+    fn v(&self) -> Parity {
+        self.parity
+    }
+
+    fn with_parity<T: Into<Parity>>(self, parity: T) -> Self {
+        Self { r: self.r, s: self.s, parity: parity.into() }
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use crate::{transaction::signature::SECP256K1N_HALF, Address, Signature, B256, U256};
     use alloy_primitives::{hex, hex::FromHex, Bytes};
     use std::str::FromStr;
 
-    #[test]
-    fn test_payload_len_with_eip155_chain_id() {
-        // Select 1 as an arbitrary nonzero value for R and S, as v() always returns 0 for (0, 0).
-        let signature = Signature { r: U256::from(1), s: U256::from(1), odd_y_parity: false };
-
-        assert_eq!(3, signature.payload_len_with_eip155_chain_id(None));
-        assert_eq!(3, signature.payload_len_with_eip155_chain_id(Some(1)));
-        assert_eq!(4, signature.payload_len_with_eip155_chain_id(Some(47)));
-    }
```

Comment:
why was this test removed?

Reply:
The corresponding method payload_len_with_eip155_chain_id() has been removed.